lib/mongoid/config/defaults.rb

```diff
@@ -39,12 +39,12 @@ def load_defaults(version)
           load_defaults "7.4"
         when "7.4"
           # flags introduced in 7.5 - old functionality
+          self.legacy_attributes = true
           self.overwrite_chained_operators = true
 
           load_defaults "7.5"
         when "7.5"
           # flags introduced in 8.0 - old functionality
-          self.legacy_attributes = true
```

Comment:
I think each branch here should set all relevant flags, in case any of them were changed previously from Mongoid's defaults, so that when an application asks for e.g. 7.5 defaults, all flags that are relevant to 7.5 are set to their default values for 7.5 regardless of existing state.

Reply:
I don't think that's necessary as the flags you're talking about are never going to get flipped back to old functionality. They are about to get removed, actually.